imap_processing/ena_maps/ena_maps.py

```diff
@@ -0,0 +1,543 @@
+    This method ensures that all pointing set pixels (and thus all counts) are
+    captured in the map, but does not ensure that all pixels in the map receive data.
+
+    **Pull Method**
+
+    The "pull" method takes each pixel in the map grid and transforms its coordinates
+    to the frame of the pointing set, then determines into which pixel in the
+    pointing set grid the transformed map pixel falls.
+    This method ensures that all pixels in the map receive data, but can result in
+    some pointing set pixels not being captured in the map, and others being captured
+    multiple times.
+    """
+
+    PUSH = "Push"
+    PULL = "Pull"
+
+
+def match_coords_to_indices(
+    spatial_object_input_frame: PointingSet | AbstractSkyMap,
+    spatial_object_output_frame: PointingSet | AbstractSkyMap,
+    event_time: float | None = None,
+) -> NDArray:
+    """
+    Find the output indices corresponding to each input coord between 2 spatial objects.
+
+    First, the pixel center coordinates of the input spatial object are
+    transformed from the Spice coordinate frame of the input object to their
+    corresponding coordinates in the Spice frame of the output object.
+    Then, the transformed pixel centers are matched to the 1D indices of the spatial
+    pixels in the output frame, either in an unwrapped rectangular grid or a Healpix
+    tessellation of the sky.
+
+    This function always "pushes" the pixels of the input object to corresponding pixels
+    in the output object's unwrapped rectangular grid or healpix tessellation;
+    however, by swapping the input and output objects, one can apply the "pull" method
+    of index  matching.
+
+    Parameters
+    ----------
+    spatial_object_input_frame : PointingSet | AbstractSkyMap
+        An object containing 1D spatial pixel centers in azimuth and elevation,
+        which will be matched to 1D indices of spatial pixels in the output frame.
+        Must contain the Spice frame in which the pixel centers are defined.
+    spatial_object_output_frame : PointingSet | AbstractSkyMap
+        The object containing a grid or tessellation of spatial pixels
+        into which the input spatial pixel centers will 'land', and be matched to
+        corresponding pixel 1D indices in the output frame.
+    event_time : float, optional
+        Event time at which to transform the input spatial object to the output frame.
+        This can be manually specified, e.g., for converting between Maps which do not
+        contain an epoch value.
+        The default value is None, in which case the event time of the PointingSet
+        object is used.
+
+    Returns
+    -------
+    flat_indices_input_grid_output_frame : NDArray
+        1D array of pixel indices of the output object corresponding to each pixel in
+        the input object. The length of the array is equal to the number of pixels in
+        the input object, and may contain 0 or >1 occurrences of the same output index.
```

Comment:
minor nit, but that seems to remove == 1. Maybe the following, or "any number of".
```suggestion
        the input object, and may contain 0, 1, or multiple occurrences of the same output index.
```